prebuilt-ui/basic-embed/components/PrebuiltCall.js

```diff
@@ -0,0 +1,254 @@
+import DailyIframe from '@daily-co/daily-js';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { writeText } from 'clipboard-polyfill';
+import { Button } from '@dailyjs/shared/components/Button';
+import {
+  Card,
+  CardBody,
+  CardFooter,
+  CardHeader,
+} from '@dailyjs/shared/components/Card';
+import Field from '@dailyjs/shared/components/Field';
+import { TextInput } from '@dailyjs/shared/components/Input';
+import { Well } from '@dailyjs/shared/components/Well';
+import { Header } from '../components/Header';
+
+export default function PrebuiltCall(props) {
+  const [demoState, setDemoState] = useState('home');
+  const [isError, setIsError] = useState(false);
+  const [roomURL, setRoomURL] = useState('');
+  const [exp, setExp] = useState();
+  const [secs, setSecs] = useState();
+  const [roomValidity, setRoomValidity] = useState(false);
+  const roomURLRef = useRef(null);
+  const iframeRef = useRef(null);
+  const callFrame = useRef(null);
+  const [linkCopied, setLinkCopied] = useState(false);
+
+  // Updates the time left that displays in the UI
+  useEffect(() => {
+    if (!exp) {
+      return false;
+    }
+    const i = setInterval(() => {
+      const timeLeft = Math.floor((new Date(exp * 1000) - Date.now()) / 1000);
+      setSecs(`${Math.floor(timeLeft / 60)}:${`0${timeLeft % 60}`.slice(-2)}`);
+    }, 1000);
+    return () => clearInterval(i);
+  }, [exp]);
+
+  // Listens for a "call" demo state, and creates then joins a callFrame as soon as that happens
+  useEffect(() => {
+    if (!iframeRef?.current || demoState !== 'call') {
+      return;
+    }
+    try {
+      callFrame.current = DailyIframe.createFrame(iframeRef.current, {
+        showLeaveButton: true,
+        iframeStyle: {
+          height: '100%',
+          width: '100%',
+          aspectRatio: 16 / 9,
+          minwidth: '400px',
+          maxWidth: '920px',
+          border: '0',
+          borderRadius: '12px',
+        },
+      });
+      callFrame.current.join({
+        url: roomURL,
+      });
+    } catch (e) {
+      setDemoState('home');
+      setIsError(true);
+      return;
+    }
+
+    const handleLeftMeeting = () => {
+      setDemoState('home');
+      setRoomValidity(false);
+      callFrame.current.destroy();
+    };
+
+    callFrame.current.on('left-meeting', handleLeftMeeting);
+  }, [demoState, iframeRef, roomURL]);
+
+  const createRoom = useCallback(
+    async (e) => {
+      if (!roomURLRef?.current.value) {
+        const roomExp = Math.round(Date.now() / 1000) + 60 * 5;
+        try {
+          const res = await fetch('/api/room', {
```

Comment:
I might move this to an `api.js` file and import the function to create a room but I didn't look at the rest of examples for reference. Feels a little nicer than having the fetch in the component directly

Reply:
The [`<CreatingRoom />`](https://github.com/daily-demos/examples/blob/main/dailyjs/basic-call/components/CreatingRoom/CreatingRoom.js) component in  `/dailyjs/basic-call` demo and our [`<PrebuiltCall />`](https://github.com/daily-co/pluot-core/blob/main/north-star/packages/docs-site/components/PrebuiltCall/PrebuiltCall.tsx) on the docs site both fetch within a component via a [Next API route](https://nextjs.org/docs/api-routes/introduction), which has to be in `/pages/api/[file name or folder]`. I'll keep as is for now since we have those other two projects that also follow the pattern, but I'll keep this feedback in mind for the future and also move the roomExp creation over to `/pages/api`.